code/game/gamemodes/wizard/artefact.dm

```diff
@@ -8,107 +8,99 @@
-					if("healing")
-						M.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/charge(null))
-						M.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/forcewall(null))
-						M.equip_to_slot_or_del(new /obj/item/gun/magic/staff/healing(M), slot_r_hand)
-						to_chat(M, "<B>Your service has not gone unrewarded, however. Studying under [H.real_name], you have learned livesaving survival spells. You are able to cast charge and forcewall.")
-					if("robeless")
-						M.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/knock(null))
-						M.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/click/mind_transfer(null))
-						to_chat(M, "<B>Your service has not gone unrewarded, however. Studying under [H.real_name], you have learned stealthy, robeless spells. You are able to cast knock and mindswap.")
 
-				M.equip_to_slot_or_del(new /obj/item/radio/headset(M), slot_l_ear)
-				M.equip_to_slot_or_del(new /obj/item/clothing/under/color/lightpurple(M), slot_w_uniform)
-				M.equip_to_slot_or_del(new /obj/item/clothing/shoes/sandal(M), slot_shoes)
-				M.equip_to_slot_or_del(new /obj/item/clothing/suit/wizrobe(M), slot_wear_suit)
-				M.equip_to_slot_or_del(new /obj/item/clothing/head/wizard(M), slot_head)
-				M.equip_to_slot_or_del(new /obj/item/storage/backpack(M), slot_back)
-				M.equip_to_slot_or_del(new /obj/item/storage/box(M), slot_in_backpack)
-				M.equip_to_slot_or_del(new /obj/item/teleportation_scroll/apprentice(M), slot_r_store)
-				var/wizard_name_first = pick(GLOB.wizard_first)
-				var/wizard_name_second = pick(GLOB.wizard_second)
-				var/randomname = "[wizard_name_first] [wizard_name_second]"
-				var/newname = sanitize(copytext(input(M, "You are the wizard's apprentice. Would you like to change your name to something else?", "Name change", randomname) as null|text,1,MAX_NAME_LEN))
-
-				if(!newname)
-					newname = randomname
-				M.mind.name = newname
-				M.real_name = newname
-				M.name = newname
-				var/datum/objective/protect/new_objective = new /datum/objective/protect
-				new_objective.owner = M:mind
-				new_objective:target = H:mind
-				new_objective.explanation_text = "Protect [H.real_name], the wizard."
-				M.mind.objectives += new_objective
-				SSticker.mode.apprentices += M.mind
-				M.mind.special_role = SPECIAL_ROLE_WIZARD_APPRENTICE
-				SSticker.mode.update_wiz_icons_added(M.mind)
-				M.faction = list("wizard")
-			else
-				used = 0
-				to_chat(H, "Unable to reach your apprentice! You can either attack the spellbook with the contract to refund your points, or wait and try again later.")
-	return
+	used = TRUE
+
+	var/image/source = image('icons/obj/cardboard_cutout.dmi', "cutout_wizard")
+	var/list/candidates = SSghost_spawns.poll_candidates("Do you want to play as the wizard apprentice of [H.real_name]?", ROLE_WIZARD, TRUE, source = source)
```

Comment:
Not sure how I feel about a proc which blocks for 10 seconds being inside of `ui_act`, since thats meant to be processed quickly. I would prefer wrapping this in its own function, which is then invoked asyncly. You can also use your `used` var to ensure no one can queue up 2 of these.